effect/src/main/scala/io/catbird/util/effect/package.scala

```diff
@@ -1,53 +1,60 @@
 package io.catbird.util
 
-import cats.effect.{ Async, ContextShift, ExitCase, IO }
+import cats.effect.{ Async, IO }
 import com.twitter.util.{ Future, Return, Throw, Try }
+
+import scala.None
 import java.lang.Throwable
 
+import cats.effect.Outcome
+import cats.effect.kernel.Resource.ExitCase
+
 import scala.util.{ Left, Right }
 
 package object effect extends FutureInstances with RerunnableInstances {
 
   /**
-   * Converts the `Future` to `F` without changing the underlying execution (same thread pool!).
+   * Converts the `Future` to `F`.
    */
   def futureToAsync[F[_], A](fa: => Future[A])(implicit F: Async[F]): F[A] = F.async { k =>
     fa.respond {
       case Return(a)  => k(Right[Throwable, A](a))
       case Throw(err) => k(Left[Throwable, A](err))
     }
+
```

Comment:
I suppose the side-effectful code in `async` should be delayed (it's not `async_`!).

Reply:
The `fa` here is already a byname-parameter, is that not sufficient?

Do we have to wrap the `F.async` call in `F.defer` or something?

Reply:
This whole block should be just:
```scala
F.async_ { k =>
  fa.respond {
    case Return(a)  => k(Right(a))
    case Throw(err) => k(Left(err))
  }
}
```

`async_` itself is implemented as 
```scala
async[A](cb => as(delay(k(cb)), None))
```